### PR TITLE
Use a LoggingScheduledExecutor

### PR DESCRIPTION
**Pull request by rozap**
* The regular one halts the subsequent runs if an exception is thrown
* This one at least logs that something went wrong

Refs EN-5238

### CODE REVIEW

File: balboa-agent/src/main/scala/com/socrata/balboa/agent/LoggingScheduledExecutor.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.slf4j.LoggerFactory
+
+class LoggingScheduledExecutor(poolSize: Int) extends ScheduledThreadPoolExecutor(poolSize) {
+  override def scheduleAtFixedRate(runnable: Runnable, delay: Long, period: Long, unit: TimeUnit): ScheduledFuture[_] = {
+    val wrapped = new LogRunnableException(runnable)
+    super.scheduleAtFixedRate(wrapped, delay, period, unit)
+  }
+
+  private class LogRunnableException(underlying: Runnable) extends Runnable {
+    val logger: Logger = Logger(LoggerFactory getLogger getClass.getName)
```
Comment (jacobsi) [inline balboa-agent/src/main/scala/com/socrata/balboa/agent/LoggingScheduledExecutor.scala]: Why not use `with StrictLogging`?

File: balboa-agent/src/main/scala/com/socrata/balboa/agent/LoggingScheduledExecutor.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    override def run(): Unit = {
+      try {
+        underlying.run()
+      } catch {
+        case e: Exception =>
+          logger.error("Underlying runnable threw an exception", e)
+      }
+    }
+  }
+}
```
Comment (jacobsi) [inline balboa-agent/src/main/scala/com/socrata/balboa/agent/LoggingScheduledExecutor.scala]: Why are you building a new class? Why isn't it sufficient to add a catch clause to the runnable in BalboaAgent?

**Comment by rozap** [comment]
Code coverage failed; I'm not sure that a sane way exists on the JVM to test log output? Redirecting stdout/stderr would be problematic for a multithreaded test execution. Also I really have no idea what the logging situation is here; there are a number of different logging backends it looks like. 

It seems like we'll have to do nastier things to get the coverage up, rather than just adding nocover to this block. 

Any thoughts?

File: balboa-agent/src/main/scala/com/socrata/balboa/agent/LoggingScheduledExecutor.scala (hunk 2 above)
Comment (rozap) [inline balboa-agent/src/main/scala/com/socrata/balboa/agent/LoggingScheduledExecutor.scala]: ok

**Comment by jacobsi** [comment]
lgmt

**Pull request merged**